examples/nlp/text_classification_with_transformer.py

```diff
@@ -0,0 +1,184 @@
+
+
+class TransformerLayer(tf.keras.layers.Layer):
+    def __init__(self, embed_dim, num_heads, ff_dim, rate=0.1):
+        super(TransformerLayer, self).__init__()
+
+        self.att = MultiHeadSelfAttention(embed_dim, num_heads)
+        self.ffn = keras.Sequential(
+            [
+                keras.layers.Dense(ff_dim, activation="relu"),
+                keras.layers.Dense(embed_dim),
+            ]
+        )
+
+        self.layernorm1 = keras.layers.LayerNormalization(epsilon=1e-6)
+        self.layernorm2 = keras.layers.LayerNormalization(epsilon=1e-6)
+
+        self.dropout1 = tf.keras.layers.Dropout(rate)
+        self.dropout2 = tf.keras.layers.Dropout(rate)
+
+    def call(self, x, training):
+        attn_output = self.att(x)
+        attn_output = self.dropout1(attn_output, training=training)
+        out1 = self.layernorm1(x + attn_output)
+
+        ffn_output = self.ffn(out1)
+        ffn_output = self.dropout2(ffn_output, training=training)
+        out2 = self.layernorm2(out1 + ffn_output)
+
+        return out2
+
+
+"""
+## Implement embedding layer
+
+Two seperate embedding layers, one for tokens, one for token index (positions).
+"""
+
+
+class EmbeddingLayer(keras.layers.Layer):
+    def __init__(self, maxlen, vocab_size, emded_dim):
+        super(EmbeddingLayer, self).__init__()
+        self.token_emb = keras.layers.Embedding(
+            input_dim=vocab_size, output_dim=emded_dim
+        )
+        self.pos_emb = keras.layers.Embedding(input_dim=maxlen, output_dim=emded_dim)
+
+    def call(self, x):
+        maxlen = tf.shape(x)[-1]
+        positions = tf.range(start=0, limit=maxlen, delta=1)
+        positions = self.pos_emb(positions)
+        x = self.token_emb(x)
+        return x + positions
+
+
+"""
+## Create classifier model using transformer layer
+
+Transformer layer outputs one vector for each time step of your input sequence. Here, we take the mean across all time steps and build a two layered feed forward network on top of it.
+"""
+
+
+class TransformerClassifier(tf.keras.Model):
+    def __init__(self, maxlen, vocab_size, embed_dim, ff_dim, num_heads):
+        super(TransformerClassifier, self).__init__()
+        self.emb = EmbeddingLayer(maxlen, vocab_size, embed_dim)
+        self.transformer = TransformerLayer(embed_dim, num_heads, ff_dim)
+        self.prehead = keras.layers.Dense(20, activation="relu")
+        self.dropout1 = keras.layers.Dropout(0.05)
+        self.dropout2 = keras.layers.Dropout(0.05)
+        self.head = keras.layers.Dense(2, activation="softmax")
+
+    def call(self, x, training):
+        x = self.emb(x)
+        x = self.transformer(x, training)
+        x = tf.math.reduce_mean(x, axis=1)
+        x = self.dropout1(x, training=training)
+        x = self.prehead(x)
+        x = self.dropout2(x, training=training)
+        x = self.head(x)
+        return x
+
+
+"""
+## Download and prepare dataset
+"""
+
+max_features = 6000  # Only consider the top 20k words
```

Comment:
6000 in this case, not 20k. I recommend using at least 15k though. 6k is too low.